assets/settings/actions.js

```diff
@@ -0,0 +1,173 @@
+import alertify from 'alertifyjs';
+import { gettext } from 'utils';
+import server from 'server';
+
+
+export const SELECT_ITEM = 'SELECT_ITEM';
+export function selectItem(id) {
+    return {type: SELECT_ITEM, id};
+}
+
+export const EDIT_ITEM = 'EDIT_ITEM';
+export function editItem(event) {
+    return {type: EDIT_ITEM, event};
+}
+
+export const NEW_ITEM = 'NEW_ITEM';
+export function newItem(data) {
+    return {type: NEW_ITEM, data};
+}
+
+export const CANCEL_EDIT = 'CANCEL_EDIT';
+export function cancelEdit(event) {
+    return {type: CANCEL_EDIT, event};
+}
+
+export const SAVE_ITEM = 'SAVE_ITEM';
+
+export const SAVE_ERROR = 'SAVE_ERROR';
+export function saveError(data) {
+    return {type: SAVE_ERROR, data};
+}
+
+export const SET_QUERY = 'SET_QUERY';
+export function setQuery(query) {
+    return {type: SET_QUERY, query};
+}
+
+export const QUERY_ITEMS = 'QUERY_ITEMS';
+export function queryItems() {
+    return {type: QUERY_ITEMS};
+}
+
+export const GET_COMPANIES = 'GET_COMPANIES';
+export function getCompanies(data) {
+    return {type: GET_COMPANIES, data};
+}
+
+export const GET_ITEMS = 'GET_ITEMS';
+export function getItems(data) {
+    return {type: GET_ITEMS, data};
+}
+
+export function updateMenu(data) {
+    return function (dispatch) {
+        dispatch(fetchItems(data.target.name));
+        dispatch(selectMenu(data));
+    };
+}
+
+export const SELECT_MENU = 'SELECT_MENU';
+export function selectMenu(data) {
+    return {type: SELECT_MENU, data};
+}
+
+function errorHandler(error, dispatch) {
+    console.error('error', error);
+
+    if (error.response.status !== 400) {
+        alertify.error(error.response.statusText);
+        return;
+    }
+    parseJSON(error.response).then(function(data) {
+        dispatch(saveError(data));
+    });
+}
+
+
+function parseJSON(response) {
+    return response.json();
+}
+
+/**
+ * Fetches users and companies for the time being
+ *
+ * @param {String} type either users or companies
+ */
+export function fetchItems(type) {
+    return function (dispatch) {
+        dispatch(queryItems());
+
+        return server.get(`/${type}/search`)
+            .then(parseJSON)
+            .then((data) => {
+                dispatch(getItems(data));
+                if (type === 'companies') {
+                    dispatch(getCompanies(data, type));
+                }
+            })
+            .catch((error) => errorHandler(error, dispatch));
+    };
+}
+
+
+function checkStatus(response) {
```

Comment:
this could probably go into the server module, maybe together with this parseJson so it would parse json on 2xx response